File: JavaScript/1048 - Aumento de Sálario.js
var salario = parseFloat(require('fs').readFileSync('/dev/stdin', 'utf8'));
var percentuais = [
    [400.00, 15],
    [800.00, 12],
    [1200.00, 10],
    [2000.00, 7],
    [Infinity, 4]
];

for (let [limite, p] of percentuais) {
    if (salario <= limite) {
        let reajuste = salario * p / 100;
        let novo = salario + reajuste;
        console.log(`Novo salario: ${novo.toFixed(2)}`);
        console.log(`Reajuste ganho: ${reajuste.toFixed(2)}`);
        console.log(`Em percentual: ${p} %`);
        break;
    }
}

File: Clojure/1048 - Aumento de Sálario.clj
(def salario (read))
(cond 
  (and (<= salario 400.00) (>= salario 0.00))
  (do
    (def reajuste (* salario 0.15))
    (def novoSalario (+ reajuste salario))
    (printf "Novo salario: %.2f\nReajuste ganho: %.2f\nEm percentual: 15 %%\n" novoSalario reajuste))

  (and (> salario 400.00) (<= salario 800.00))
  (do
    (def reajuste (* salario 0.12))
    (def novoSalario (+ reajuste salario))
    (printf "Novo salario: %.2f\nReajuste ganho: %.2f\nEm percentual: 12 %%\n" novoSalario reajuste))

  (and (> salario 800.00) (<= salario 1200.00))
  (do
    (def reajuste (* salario 0.10))
    (def novoSalario (+ reajuste salario))
    (printf "Novo salario: %.2f\nReajuste ganho: %.2f\nEm percentual: 10 %%\n" novoSalario reajuste))

  (and (> salario 1200.00) (<= salario 2000.00))
  (do
    (def reajuste (* salario 0.07))
    (def novoSalario (+ reajuste salario))
    (printf "Novo salario: %.2f\nReajuste ganho: %.2f\nEm percentual: 7 %%\n" novoSalario reajuste))

  :else
  (do
    (def reajuste (* salario 0.04))
    (def novoSalario (+ reajuste salario))
    (printf "Novo salario: %.2f\nReajuste ganho: %.2f\nEm percentual: 4 %%\n" novoSalario reajuste)))
